src/internal_flash/internal_flash_maz_v0.c

```diff
@@ -1264,8 +1264,61 @@ NETX_CONSOLEAPP_RESULT_T internal_flash_maz_v0_read(CMD_PARAMETER_READ_T *ptPara
 #       if CFG_INCLUDE_SHA1!=0
 NETX_CONSOLEAPP_RESULT_T internal_flash_maz_v0_sha1(CMD_PARAMETER_CHECKSUM_T *ptParameter, SHA_CTX *ptSha1Context)
 {
-	uprintf("! not yet....\n");
-	return NETX_CONSOLEAPP_RESULT_ERROR;
+	NETX_CONSOLEAPP_RESULT_T tResult;
+	const INTERNAL_FLASH_ATTRIBUTES_MAZ_V0_T *ptAttr;
+	unsigned long ulOffsetStart;
+	unsigned long ulOffsetEnd;
+	unsigned long ulLength;
+	const unsigned char *pucFlashArea;
+	const unsigned char *pucFlashStart;
+	FLASH_BLOCK_ATTRIBUTES_T tFlashBlock;
+
+
+	ulOffsetStart = ptParameter->ulStartAdr;
+	ulOffsetEnd = ptParameter->ulEndAdr;
+
+	/* Be pessimistic. */
+	tResult = NETX_CONSOLEAPP_RESULT_ERROR;
+
+	uprintf("# Calculating checksum...\n");
+
+	/* Silently ignore read requests with a size of 0 bytes. */
+	if( ulOffsetStart==ulOffsetEnd )
+	{
+		tResult = NETX_CONSOLEAPP_RESULT_OK;
+	}
+	else
+	{
+		/* Get a pointer to the flash attributes. */
+		ptAttr = &(ptParameter->ptDeviceDescription->uInfo.tInternalFlashInfo.uAttributes.tMazV0);
+
+		tResult = check_command_area(ptAttr, ulOffsetStart, ulOffsetEnd);
+		if( tResult==NETX_CONSOLEAPP_RESULT_OK )
+		{
+			/* Get the pointer to the controller and the offset in the memory map. */
+			tResult = iflash_get_controller(ptAttr, ulOffsetStart, &tFlashBlock);
```

Comment:
If the "virtual" flash unit no. 3 is used (= units 0 and 1 combined), this only configures unit 0. 
We may have to configure unit 1 as well to prevent reading invalid data. This can occur if
- flash unit 3 is selected,
- the start offset is in unit 0 
- and the end offset is in unit 1.
See the remark in internal_flash_maz_v0_read() ("If Iflash01 (unit 3) is selected...")